arch/arc/core/cache.c

```diff
@@ -67,34 +67,19 @@ static void dcache_enable(void)
 	dcache_dc_ctrl(DC_CTRL_DC_ENABLE);
 }
 
-
-/**
- *
- * @brief Flush multiple d-cache lines to memory
- *
- * No alignment is required for either <start_addr> or <size>, but since
- * dcache_flush_mlines() iterates on the d-cache lines, a cache line
- * alignment for both is optimal.
- *
- * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
- * kconfig option or it is detected at runtime.
- *
- * @param start_addr the pointer to start the multi-line flush
- * @param size the number of bytes that are to be flushed
- *
- * @return N/A
- */
-static void dcache_flush_mlines(uint32_t start_addr, uint32_t size)
+void arch_dcache_flush(void *start_addr, size_t size)
 {
-	uint32_t end_addr;
+	uintptr_t start_addr_temp = (uintptr_t)start_addr;
+	uintptr_t end_addr;
```

Comment:
Just a suggestion: it would be more straight forward if we rename `void *start_addr` to `start_addr_ptr` and `uintptr_t start_addr_temp` to simply `start_addr`.

Reply:
Renaming start_addr  to start_addr_ptr  is okay. But we needed a local variable start_addr_temp  to typecast the void* start_addr to uintptr_t . So that typecasting is not done everywhere with start_addr. Let me know if you have any suggestions here given we would need a local variable whether it be start_addr_temp or any other name. Thanks

Reply:
I think @stephanosio is suggesting:
```
void arch_dcache_flush(void *start_addr_ptr, size_t size)
{
uintptr_t start_addr = (uintptr_t)start_addr_ptr;
...
Now you can use start_addr everywhere in the function instead of start_addr_temp
...
}
```